doc/source/administrator/optimization.md

```diff
@@ -383,3 +383,133 @@ Choosing the right _limit to guarantee ratio_ ratio is an art, not a science.
 We suggest **starting with a ratio of 2 to 1** and adjusting from there based on whether you run into problems on your hub.
 For example, if the limit is 10GB, start with a guarantee of 5GB.
 Use a service such as Prometheus + Grafana to monitor the memory usage over time, and use this to decide whether to adjust your ratio.
+
+## Explicit memory and CPU allocated to core pods' containers
+
+The Helm chart creates several k8s Pods that typically have one container
+running in them, but sometimes more. In this section you are guided on how to
+explicitly specify how much CPU and memory they are guaranteed via _requests_
+and how much CPU and memory they are limited to via _limits_.
+
+To decide what requests and limits to set, some background knowledge is
+relevant:
+
+1. A [LimitRange](https://kubernetes.io/docs/concepts/policy/limit-range/)
+   resource in a namespace can provide defaults to containers without explicitly
+   set requests and limits.
+
+   Check the namespace where you deploy JupyterHub for such resource by doing
+   `kubectl get limitrange --namespace <k8s-namespace>`.
+
+1. Setting only resource limits _will make_ k8s assume the same in resource
+   requests, but setting only resource requests _will not make_ k8s assume any
+   resource limits.
+
+1. Setting resource requests improves the reliability of the container's
+   execution. Setting resource limits improves the consistency of the execution
```

Comment:
Yeah hahah i figured when I wrote that that it would be not so obvious what i meant, but I felt that it would be troublesome to explain it in detail as well.

By limiting your pod to a certain memory for example, you can consistently make it crash by trying to surpass the limit. But, for a higher reliability of the containers execution the limit could be removed. It is then less prone to crash due to memory limits, but less consistent in when it does.

Reply:
I attempted a new formulation in 1a7bc6d

> 1. Setting resource _requests_ guarantees a minimal level of resources for the
   container and therefore some minimal capacity for it to do its job. Setting
   resource _limits_ close to or equal to the resource requests can be useful to
   make the container operational performance more consistent.